### ISSUE TITLE
Run-time error for searches with just an exclusion term (with a solution)

### ISSUE THREAD

**Issue by Leu-Gim** [nim-lang/nimforum#87]
Searches like "-qwe" result in a run-time error (exception).
This due to ``-`` before a word stands for exclusion (for no such word in results) in SQLite's full-text search, and it disallows using only exclusions in a search. Though I expected just an empty result from such a search, not an exception.

Such code replacing line 1338 in forum.nim solves the problem:

```nim
    var q = @"q".strip
    if q[0]=='-' and not q.contains(Whitespace):
      q=""
```

If a PR's needed, I can make it, or the code can be taken from here.

**Comment by Leu-Gim** [comment]
There still remain many ways to raise an error with that code, so it's better instead of that to add exception handling for SQLite calls: to replace (lines 1348-1350)

```nim
      for rowFT in fastRows(db, queryFT,
                    [q,q,$ThreadsPerPage,$c.pageNum,$ThreadsPerPage,q,
                     q,q,$ThreadsPerPage,$c.pageNum,$ThreadsPerPage,q]):
```

with

```nim
      let rows =
            try: getAllRows(db, queryFT,
                    [q,q,$ThreadsPerPage,$c.pageNum,$ThreadsPerPage,q,
                     q,q,$ThreadsPerPage,$c.pageNum,$ThreadsPerPage,q])
            except: @[]
      for rowFT in rows:

**Comment by markprocess** [comment]
@Leu-Gim can you reproduce this in the current deployment? I was unable to. Regards, Mark. cc @dom96

**Comment by Leu-Gim** [comment]
After transition to Karax (the new forum) `-` don't work any more at all, as I see,
and "-someword" is treated as just "someword".

If it worked, the issue would not be visible to users, I think,
because results are fetched via ajax,
via subrequests to ``/search.json?q=...``,
and that exception would raise in that request,
and at client side there's a check for HTTP status of the ajax request prior to processing results.

**Comment by Leu-Gim** [comment]
Say "-nick" returns results for "nick"...

**Comment by Leu-Gim** [comment]
Oh, Forum has been moved to SQLite's "Enhanced Query Syntax" (https://sqlite.org/fts3.html) now, as I see - `AND`, `OR` and parrentheses work. Then `NOT` should work instead of `-`.

**Comment by Leu-Gim** [comment]
Yes, `NOT` does work too. An example query with all three operators and `()` for test: "(csp AND oberon) OR (actionscript NOT mtasc)" (picked for obviousness).

Queries with just negation (``NOT word``) still fail, and the server doesn't work then for a couple of seconds. So the issue is the same, with just `NOT ` instead of `-`. And solution is the same.

**Comment by markprocess** [comment]
@Leu-Gim Thanks - I will study this soon. Mark.
